trade-finance-manager-api/src/v1/rest-mappings/mappings/gef-facilities/mapGefFacilitySnapshot.js

```diff
@@ -0,0 +1,64 @@
+const { formattedNumber } = require('../../../../utils/number');
+const mapFacilityValue = require('../facilities/mapFacilityValue');
+const mapFacilityProduct = require('../facilities/mapFacilityProduct');
+const mapFacilityType = require('../facilities/mapFacilityType');
+const mapGuaranteeFeePayableToUkef = require('../facilities/mapGuaranteeFeePayableToUkef');
+const mapGefUkefFacilityType = require('./mapGefUkefFacilityType');
+const mapGefFacilityDates = require('./mapGefFacilityDates');
+const mapFacilityValueExportCurrency = require('../facilities/mapFacilityValueExportCurrency');
+const mapUkefExposureValue = require('../facilities/mapUkefExposureValue');
+const { mapGefFacilityStage } = require('../facilities/mapFacilityStage');
+
+/**
+ * Maps the existing GEF facility snapshot in the database to the facility snapshot used in TFM API.
+ * This function is only used on GEF facilities.
+ * Note: This implementation modifies the facility snapshot to have values not consistent with the facility snapshot in the database.
+ * In particular, this is a live object that updates e.g. when amendments are added in TFM.
+ * @param facility the full facility object from the database
+ * @param dealSnapshot the deal.dealSnapshot object from the database corresponding to the facility
+ * @returns mapped facility snapshot for use in TFM
+ */
+const mapGefFacilitySnapshot = (facility, dealSnapshot) => {
+  const { facilitySnapshot, tfm: facilityTfm } = facility;
+
+  const { dealId, coverPercentage, currency, value, interestPercentage, feeType, feeFrequency, hasBeenIssued, name, type, ukefFacilityId, guaranteeFee } =
+    facilitySnapshot;
+
+  const formattedFacilityValue = formattedNumber(value);
+
+  facilitySnapshot.facilityProduct = mapFacilityProduct(type);
+
+  facilitySnapshot.facilityStage = mapGefFacilityStage(hasBeenIssued);
+
+  facilitySnapshot.ukefFacilityType = type;
+
+  return {
```

Comment:
Can we have a `const mapped` please - similar comment as above

Reply:
Have changed!